sssd_test_framework/roles/ad.py

```diff
@@ -1733,43 +1735,53 @@ def add(self) -> GPO:
 
     def link(
         self,
-        op: str | None = "New",
         target: str | None = None,
-        args: list[str] | str | None = None,
+        enforced: bool | None = None,
+        disabled: bool | None = False,
+        order: int | None = 0,
     ) -> GPO:
         """
-        Link the group policy to the a target object inside the directory, a site, domain or an ou.
-
-        ..Note::
-            The New and Set cmdlets are identical. To modify an an existing link,
-            change the $op parameter to "Set", i.e. to disable 'Enforced'
+        Link the group policy to the target object inside the directory, a site, domain or an ou.
 
-            ou_policy.link("Set", args=["-Enforced No"])
-
-        :param op: Cmdlet operation, defaults to "New"
-        :type op: str, optional
         :param target: Group policy target
         :type target: str, optional
-        :param args: Additional arguments
-        :type args: list[str] | None, optional
+        :param enforced: Enforced the policy
+        :type enforced: bool, optional
+        :param disabled: Disable the policy
+        :type disabled: bool, optional
+        :param order: Order number
+        :type order: int, optional
         :return: Group policy object
         :rtype: GPO
         """
-        if args is None:
-            args = []
+        args: str = ""
 
-        if isinstance(args, list):
-            args = " ".join(args)
-        elif args is None:
-            args = ""
+        if enforced is True:
+            args = args + " -Enforce Yes"
+        if enforced is False:
+            args = args + " -Enforce No"
+
+        if disabled is True:
+            args = args + " -LinkEnabled No"
+        else:
+            args = args + " -LinkEnabled Yes"
+
+        if order != 0:
+            args = args + f" -Order {str(order)}"
```

Comment:
You probably want to use the cli builder class? E.g. https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/roles/ad.py#L620